### DISCUSSION TITLE
Style JSON code with comments?

### DISCUSSION THREAD

**Discussion by Sverre-Iversen** [ldproxy/ldproxy#854] [Configuration] [answered]
**It is possible to enable comments in the json code for styling?**
like 
//comment 
or 
/* 
Comment
*/
I see that a "comment"-property for the layer object works, but it is often not enough

Kind Regards
Sverre

**Comment by azahnen** [comment]
Since JSON does not support comments, the answer unfortunately is a firm no.

**Reply by Sverre-Iversen** [reply]
This comment technique will work in JSON Style:
...
"comments": [ "Comment 1...",  "Comment 2...",  "Comment n..." ],
...

**Comment by cportele** [comment, marked as answer]
Adding additional elements like "comments" in any object of the style should work, at least when the style is manually placed in the "api-resources" folder. 

However, when the style is accessed via the API, the "comments" member will no longer be included. The same is true, if a style is created/replaced via the API using POST/PUT. The reason is that for validation and security purposes the style content is de-serialized into a Java object following the style schema - and that does not include "comments" members.

If the styles are not submitted via the API and the comments are only for those editing the styles, would adding additional members as needed work for your use case? If not, in what objects would you need a comments member in addition to layers? 

What can always be included is a top-level "metadata" member with any object content. This will be visible when the style is accessed via the API.

**Reply by Sverre-Iversen** [reply]
\>What can always be included is a top-level "metadata" member with any object content.
\>This will be visible when the style is accessed via the API.
Perfect, I have started using the "metadata"-property at the top-level now 👍 (in addition to a  non-visible "comments"-property)

\>...would adding additional members as needed work for your use case?
Yes, adding a non-visible "comments"-property will work for me at the "layer"-level. 
However it would be **convenient to have an extra  visible "description"-property** (or similar) in the "layer"- & "sources"-objects when the style is accessed via the API. The "id"-property is in its nature not very descriptive.
